nightfall-deployer/contracts/Challenges.sol

```diff
@@ -224,31 +240,36 @@ contract Challenges is Stateful, Key_Registry, Config {
 
         // now we need to check that the proof is correct
         ChallengesUtil.libChallengeProofVerification(
-            transactions[transactionIndex],
+            transaction.transaction,
             extraPublicInputs,
             uncompressedProof,
-            vks[transactions[transactionIndex].transactionType]
+            vks[transaction.transaction.transactionType]
         );
-        challengeAccepted(blockL2);
+        challengeAccepted(transaction.blockL2);
     }
 
-    function challengeHistoricRoot(
-        Block calldata blockL2,
-        Transaction[] calldata transactions,
-        uint256 transactionIndex
-    ) external onlyBootChallenger {
+    function challengeHistoricRoot(TransactionInfoBlock calldata transaction)
```

Comment:
I know this wasn't part of the PR, but can we change this to `challengeHistoricRootBlockNumber` - there  might a reference that has to be updated in `optimist`